packages/store/ts/library/config/index.ts

```diff
@@ -1,21 +1,29 @@
 import { AbiType, AbiTypes, StaticAbiType, StaticAbiTypes } from "@latticexyz/schema-type";
 import { RefinementCtx, z, ZodIssueCode } from "zod";
-import { AsDependent, ExtractUserTypes, RequireKeys, StaticArray, StringForUnion } from "@latticexyz/common/type-utils";
+import {
+  AsDependent,
+  ExtractUserTypes,
+  OrDefaults,
+  RequireKeys,
+  StaticArray,
+  StringForUnion,
+} from "@latticexyz/common/type-utils";
 import {
   // validation utils
   getDuplicates,
   parseStaticArray,
   STORE_SELECTOR_MAX_LENGTH,
   // config
   MUDCoreUserConfig,
-  MUDCoreConfig,
   mudCoreConfig,
   // schemas
   zObjectName,
   zSelector,
   zUserEnum,
   zValueName,
 } from "@latticexyz/config";
+// typeExtensions are necessary for `ExpandMUDUserConfig`, but configExtensions can be avoided
+import "../../register/typeExtensions";
```

Comment:
could you provide some more context for why this is needed? feels a little wrong to have the separation into `library` and `register` but then also have global type side effects in `library`

Reply:
removed the need for it by moving `mudConfig` and `ExpandMUDUserConfig` to `/register` export. It was needed because `mudConfig`, while expanding strong types, makes assumptions about keys in `MUDUserConfig` which only work locally or if `typeExtensions` are imported.
For example running type tests in `cli` (which doesn't import `/register` anywhere) would break those assumptions. But things that import `/library` shouldn't really need `mudConfig` or the global expander, so moving those to `/register` seems like the best solution